### DISCUSSION TITLE
Debugging

### DISCUSSION THREAD

**Discussion by ErikGrav** [PrivateGER/patreon-dl#16] [General]
Apologies, stupid question, but just poking at GO for the first time here. 

How do you go about setting up the debug config? 
I assume its something like this. 
but the three arguments that go in.. what is expeceted to run it? 
ServeGadget, UserInfo, DownloadURLCollector
I can compile it all fine, runs fine once compiled. But its rather...  counterproductive way of doing it. 

Any chanse you could help me out getting it running? 
Trying to look into adding more filtering options. Such as filtering on tag names. 

```
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Launch",
            "type": "go",
            "request": "launch",
            "mode": "auto",
            "program": "Main.go",
            "env": {},
            "args": []
        }
    ]
}
```

**Comment by PrivateGER** [comment]
The functions you listed are HTTP response functions, meaning that they're more of a pointer to a function. They implement an interface and have a signature that looks like this:
`func(ResponseWriter, *Request)`

They're called when the relevant endpoint is hit. Debugging is better done deeper inside.

**Reply by ErikGrav** [reply]
Thank you. 

Thought perhaps it was possible like in rust: just set up a debug config and run it and be able to step through the code. 
I'll look a bit more into it tomorrow as to how one can run go programs. 


The reason I asked was: 
I try to manually run it. 
```
 go run main.go
# command-line-arguments
.\main.go:12:29: undefined: ServeGadget
.\main.go:13:27: undefined: UserInfo
.\main.go:14:31: undefined: DownloadURLCollector
```

Im doing something wrong, il figure it out :) 
https://github.com/golang/vscode-go/wiki/debugging

Seem to be the way to go

**Reply by PrivateGER** [reply]
go run doesn't work because this is a multi-file program. 

You want something go build based.
